helm/cluster-openstack/templates/openstack_machine_template.yaml

```diff
@@ -1,14 +1,14 @@
-{{- range (append .Values.nodeClasses ((merge (dict "name" "control-plane") .Values.controlPlane))) }}
+{{- range $name, $value := (merge .Values.nodeClasses (dict "controlPlane" .Values.controlPlane)) }}
```

Comment:
`control-plane` -> `controlPlane` is a breaking change here.

Error: OpenStackMachineTemplate.infrastructure.cluster.x-k8s.io "valu1-controlPlane-91da4cd9" is invalid: metadata.name: Invalid value: "valu1-controlPlane-91da4cd9": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')